### ISSUE TITLE
AXIS Adapter (Axis Camera Application Platform - ACAP)

### ISSUE THREAD

**Issue by dalig83** [ioBroker/AdapterRequests#571]
**What kind of device or service you would like to see an adapter for?**
AXIS Cameras, AXIS Network Video Intercom, AXIS Network speakers, ...

The ACAP SDK provides:

- Full access to most common product features, including video, audio, the event system, I/O ports, PTZ control, and more.
- Enables hardware acceleration on specific functions such as image analysis computations, overlay graphics and more.
- Support for multiple architectures to build applications for all ACAP-enabled Axis products.
- Example applications for all APIs to kickstart your development.

**Is an official API including documentation is available?**
Yes, there is
https://www.axis.com/techsup/developer_doc/acap3/3.2/main/html/index.html
https://www.axis.com/products/online-manual/s00004#t10160862

There are some examples/tutorials already on GitHub
https://github.com/AxisCommunications/acap3-examples

**Are other libraries for an integration available?**
Unknown

**Is this device already integrated in other Smart Home systems?**
Unknown

**Is this device already integrated in homebridge? Might the ham adapter together with the homebridge plugin be sufficient?**
Unknown

**Additional context**
Some ideas of what could be done with this adapter:

- receive/send audio in different formats from/to camera
- receive images from camera
- receive video stream from camera
- overlay the video stream with specific information
- send and receive events from/to camera
- access stored data from SD card or NAS
- control pan/tilt/zoom 
- ...

**Comment by dalig83** [comment]
@Apollon77 does the tag "Hardware needed" mean you will implement it, when you get the hardware?
Sorry, I am not familiar with the meaning of the tags yet.

**Comment by Apollon77** [comment]
No, it is more the generic "any dev that find time and be interested in tat would need hardware to do it"

**Comment by dalig83** [comment]
OK, thanks, understood. I will try to find a way to get HW.

**Comment by dalig83** [comment]
AXIS offers a "Virtual loan tool"

> Get online access to an Axis product to run integration compatibility tests and product tests. The subscription includes full access to the product for a day, and you can upload ACAP applications, take control of PTZ functions, etc. Due to port forwarding limitations, video can only be streamed using RTSP over HTTP, or using Multipart MJPEG.

> The number of products available in the loan tool is limited. Please return the product as soon as you are finished with it.

> Once the product has been returned it will automatically be reset to factory default settings and re-appear in the list of available products. This process may take up to 10 minutes.

Would this also help?
